src/controllers/AddressToRemoveController.ts

```diff
@@ -0,0 +1,41 @@
+import { NextFunction, Request, Response } from 'express';
+import { StatusCodes  } from 'http-status-codes';
+
+import { Address } from '../models/SuppressionDataModel'
+import { ADDRESS_TO_REMOVE_PAGE_URI } from '../routes/paths';
+import SessionService from '../services/SessionService'
+import { ValidationResult } from '../utils/validation/ValidationResult';
+import { FormValidator } from '../validators/FormValidator';
+import { schema as formSchema } from '../validators/schema/AddressToRemoveSchema'
+
+const template = 'address-to-remove';
+
+export class AddressToRemoveController {
+
+  constructor(private validator: FormValidator = new FormValidator(formSchema)) {}
+
+  public renderView = (req: Request, res: Response, next: NextFunction) => {
+    const session = SessionService.getSuppressionSession(req);
+    res.render(template, { ...session?.addressToRemove });
+  }
+
+  public processForm = async (req: Request, res: Response, next: NextFunction) => {
+    const session = SessionService.getSuppressionSession(req);
+    if (!session) {
+      res.status(StatusCodes.INTERNAL_SERVER_ERROR).render('error');
+    } else {
+      const validationResult: ValidationResult = await this.validator.validate(req);
+      if (validationResult.errors.length > 0) {
+        res.status(StatusCodes.UNPROCESSABLE_ENTITY);
+        return res.render(template, {
+          ...req.body,
+          validationResult
+        });
+      } else {
+        session.addressToRemove = req.body as Address;
+        SessionService.setSuppressionSession(req, session);
+        res.redirect(ADDRESS_TO_REMOVE_PAGE_URI);
+      }
+    }
```

Comment:
we could prob decrease some of this nesting, what do you think

```suggestion
    if (!session) {
      res.status(StatusCodes.INTERNAL_SERVER_ERROR).render('error');
    } 
      
    const validationResult: ValidationResult = await this.validator.validate(req);
    if (validationResult.errors.length > 0) {
      res.status(StatusCodes.UNPROCESSABLE_ENTITY);
      return res.render(template, {
        ...req.body,
        validationResult
      });
    } else {
      session.addressToRemove = req.body as Address;
      SessionService.setSuppressionSession(req, session);
      res.redirect(ADDRESS_TO_REMOVE_PAGE_URI);
      }
    }
```

Reply:
I did initially try this but res.status() isn't a terminal. I'll have a fiddle with returns and see if I can get it to work.

Reply:
I comment this later, but this error template is out of scope anyway

Reply:
Responded to the comments about the error page. Managed to decrease nesting by returning the `res.render...`.